### PR TITLE
Add README for storage blob samples

### PR DESCRIPTION
**Pull request by kyle-patterson**
This PR introduces a README to the Storage-Blobs sample directory

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+page_type: sample
+languages:
+- .NET
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: dotnet, not .NET

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- azure-storage
+---
+
+# Azure Storage Blob client library samples for .NET
+
+These sample programs show how to use the .NET client libraries for Azure Storage Blobs in some common scenarios.
+
+|__File Name__|__Description__|
+|-------------|---------------|
+|[Sample01a_HelloWorld](Sample01a_HelloWorld.cs)|upload, download and list blobs|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: Please turn all of the descriptions into sentences with proper casing and punctuation.

Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: download,

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+|[Sample01b_HelloWorldAsync](Sample01b_HelloWorldAsync.cs)|asynchronously upload, download and list blobs|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: download,

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+|[Sample02_Auth](Sample02_Auth.cs)|authenticate with connection strings, public access, shared keys, shared access signatures, and Azure Active Directory|
+|[Sample03a_Batching](Sample03a_Batching.cs)|Upload, download, delete and set access policies for blobs in batches|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: delete,

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+|[Sample03b_BatchingAsync](Sample03b_BatchingAsync.cs)|asynchronously upload, download, delete and set access policies for blobs in batches|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: delete,

**Review by jongio — CHANGES_REQUESTED** [review]

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
 products:
 - azure
 - azure-storage
 ---
 
-# Azure Storage Blob client library samples for .NET
+# Azure Storage Blob client library samples for dotnet
 
-These sample programs show how to use the .NET client libraries for Azure Storage Blobs in some common scenarios.
+These sample programs show how to use the dotnet client libraries for Azure Storage Blobs in some common scenarios.
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: .NET should be used here, not dotnet.  The slug in the front-matter needs to be dotnet

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
-|[Sample01b_HelloWorldAsync](Sample01b_HelloWorldAsync.cs)|asynchronously upload, download and list blobs|
-|[Sample02_Auth](Sample02_Auth.cs)|authenticate with connection strings, public access, shared keys, shared access signatures, and Azure Active Directory|
-|[Sample03a_Batching](Sample03a_Batching.cs)|Upload, download, delete and set access policies for blobs in batches|
-|[Sample03b_BatchingAsync](Sample03b_BatchingAsync.cs)|asynchronously upload, download, delete and set access policies for blobs in batches|
\ No newline at end of file
+|[Sample01a_HelloWorld](Sample01a_HelloWorld.cs)|Samples demonstrating how to upload, download, and list blobs.|
+|[Sample01b_HelloWorldAsync](Sample01b_HelloWorldAsync.cs)|Samples demonstrating how to asynchronously upload, download, and list blobs.|
+|[Sample02_Auth](Sample02_Auth.cs)|Samples demonstrating how to authenticate with connection strings, public access, shared keys, shared access signatures, and Azure Active Directory.|
+|[Sample03a_Batching](Sample03a_Batching.cs)|Samples demonstrating how to upload, download, delete, and set access policies for blobs in batches|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: Missing period.

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 8]
```diff
@@ -1,20 +1,20 @@
...
+|[Sample03b_BatchingAsync](Sample03b_BatchingAsync.cs)|Samples demonstrating how to asynchronously upload, download, delete, and set access policies for blobs in batches|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: Missing period.

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 9]
```diff
@@ -1,20 +1,20 @@
...
 
 |__File Name__|__Description__|
 |-------------|---------------|
-|[Sample01a_HelloWorld](Sample01a_HelloWorld.cs)|upload, download and list blobs|
-|[Sample01b_HelloWorldAsync](Sample01b_HelloWorldAsync.cs)|asynchronously upload, download and list blobs|
-|[Sample02_Auth](Sample02_Auth.cs)|authenticate with connection strings, public access, shared keys, shared access signatures, and Azure Active Directory|
-|[Sample03a_Batching](Sample03a_Batching.cs)|Upload, download, delete and set access policies for blobs in batches|
-|[Sample03b_BatchingAsync](Sample03b_BatchingAsync.cs)|asynchronously upload, download, delete and set access policies for blobs in batches|
\ No newline at end of file
+|[Sample01a_HelloWorld](Sample01a_HelloWorld.cs)|Samples demonstrating how to upload, download, and list blobs.|
```
Comment (jongio) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: You can remove "Samples demonstrating" and start with "How to"

**Review by jongio — CHANGES_REQUESTED** [review]

**Review by jongio — APPROVED** [review]
Looks good!

**Comment by JoshLove-msft** [comment]
Looks like CI is failing due to the Readme checks. I think these checks for the various headers like Getting Started, Key Concepts, etc shouldn't apply to sample directory Readme. @weshaggard any chance we can exclude these across the board, or should we just update the eng/.docsettings.yml file to ignore these readmes?

**Comment by jongio** [comment]
Can you share the CI failure message?

**Comment by JoshLove-msft** [comment]
> Can you share the CI failure message?

https://dev.azure.com/azure-sdk/public/_build/results?buildId=198645&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=9069f0a2-bd31-51b1-8b5c-08355cf80229

You can also get to it by clicking Details link next to net-storage-ci.

**Comment by jongio** [comment]
IC, the build process verifies that README files have certain sections in them.  

This README isn't a standard readme, it is one that will be used by the docs/samples app.

I recommend excluding READMEs under /samples folders, as I don't think they will follow this format going forward.

```
\sdk\storage\Azure.Storage.Blobs\samples\README.md is missing headers with patterns:
 * Azure .+ client library for .NET
 * ^Getting started$
 * ^Key concepts$
 * ^Examples$
 * ^Troubleshooting$
 * ^Next steps$
 * ^Contributing$

```

**Comment by weshaggard** [comment]
>Looks like CI is failing due to the Readme checks. I think these checks for the various headers like Getting Started, Key Concepts, etc shouldn't apply to sample directory Readme. @weshaggard Wes Haggard FTE any chance we can exclude these across the board, or should we just update the eng/.docsettings.yml file to ignore these readmes?

For now please add to the exclude list but @chidozieononiwu is going to be working on a more global exclude option for any readme that doesn't belong to a package.

File: sdk/storage/Azure.Storage.Blobs/samples/README.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+---
+page_type: sample
+languages:
+- dotnet
+products:
+- azure
+- azure-storage
+---
+
+# Azure Storage Blob client library samples for .NET
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/samples/README.md]: Nit - consider changing to `Azure.Storage.Blobs client library samples`.

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/samples/README.md (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/samples/README.md]: `dotnet` is a product slug, not [one of the devlangs](https://review.docs.microsoft.com/en-us/new-hope/information-architecture/metadata/taxonomies?branch=master#dev-lang).  We should use `csharp` instead.

**Comment by tg-msft** [comment]
I've included these changes in #8968 which adds them for the rest of the Track 2 libraries

**Comment by kyle-patterson** [comment]
> 
> 
> I've included these changes in #8968 which adds them for the rest of the Track 2 libraries

Got it, thanks! If there are no objections, I'll close this PR.

**Review by amnguye — APPROVED** [review]

**Comment by AlexGhiondea** [comment]
@kyle-patterson could you rebase this?

**Comment by tg-msft** [comment]
I'm just closing this - we've already included this change.

**Pull request closed without merging**